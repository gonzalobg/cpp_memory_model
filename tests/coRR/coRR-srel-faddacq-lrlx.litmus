C coRR-srel-faddacq-lrlx
{ [x] = 0; }

P0 (int* x) {
  atomic_store_explicit(x, 1, memory_order_release);
}

P1 (int* x) {
  int a = atomic_fetch_add_explicit(x, 0, memory_order_acquire);
  if (a == 1) {
    int b = atomic_load_explicit(x, memory_order_relaxed);
  }
}

~exists (1:a=1 /\ 1:b=0)