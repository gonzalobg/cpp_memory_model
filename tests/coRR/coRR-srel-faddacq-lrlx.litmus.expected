Test coRR-srel-faddacq-lrlx Forbidden
States 2
1:a=0; 1:b=0;
1:a=1; 1:b=1;
Ok
Witnesses
Positive: 2 Negative: 0
Condition ~exists (1:a=1 /\ 1:b=0)
Observation coRR-srel-faddacq-lrlx Never 0 2
Hash=fc6bf1ae1e651254a9540ef515b0640e

